_data-prepper/pipelines/configuration/processors/flatten.md

```diff
@@ -0,0 +1,239 @@
+{
+  "key1": "val1",
+  "key2": {
+    "key3": {
+      "key4": "val2"
+    }
+  }
+}
+```
+
+The `flatten` processor creates a flattened structure under the `flattened-key2` object, as shown in the following output:
+
+```json
+{
+  "key1": "val1",
+  "key2": {
+    "key3": {
+      "key4": "val2"
+    }
+  },
+  "flattened-key2": {
+    "key3.key4": "val2"
+  }
+}
+```
+
+### Remove processed fields
+
+Use the `remove_processed_fields` option when flattening all of an event's nested objects. This removes all the event's processed fields, as shown in the following example:
+
+```yaml
+...
+  processor:
+    - flatten:
+        source: ""   # empty string represents root of event
+        target: ""   # empty string represents root of event
+        remove_processed_fields: true
+...
+```
+
+For example, when the input event contains the following nested objects:
+
+```json
+{
+  "key1": "val1",
+  "key2": {
+    "key3": {
+      "key4": "val2"
+    }
+  },
+  "list1": [
+    {
+      "list2": [
+        {
+          "name": "name1",
+          "value": "value1"
+        },
+        {
+          "name": "name2",
+          "value": "value2"
+        }
+      ]
+    }
+  ]
+}
+```
+
+
+The `flatten` processor creates a flattened structure where all processed fields are absent, as shown in the following output:
+
+```json
+{
+  "key1": "val1",
+  "key2.key3.key4": "val2",
+  "list1[0].list2[0].name": "name1",
+  "list1[0].list2[0].value": "value1",
+  "list1[0].list2[1].name": "name2",
+  "list1[0].list2[1].value": "value2",
+}
+```
+
+### Exclude specific keys from flattening
+
+Use the `exclude_keys` option to keep specific keys from being flattened in the output, as shown in the following example where the `key2` value is excluded:
+
+```yaml
+...
+  processor:
+    - flatten:
+        source: ""   # empty string represents root of event
+        target: ""   # empty string represents root of event
+        remove_processed_fields: true
+        exclude_keys: ["key2"]
+...
+```
+
+For example, when the input event contains the following nested objects:
+
+```json
+{
+  "key1": "val1",
+  "key2": {
+    "key3": {
+      "key4": "val2"
+    }
+  },
+  "list1": [
+    {
+      "list2": [
+        {
+          "name": "name1",
+          "value": "value1"
+        },
+        {
+          "name": "name2",
+          "value": "value2"
+        }
+      ]
+    }
+  ]
+}
+```
+
+All other nested objects in the input event excluding the `key2` key will be flattened, as shown in the following example:
+
+```json
+{
+  "key1": "val1",
+  "key2": {
+    "key3": {
+      "key4": "val2"
+    }
+  },
+  "list1[0].list2[0].name": "name1",
+  "list1[0].list2[0].value": "value1",
+  "list1[0].list2[1].name": "name2",
+  "list1[0].list2[1].value": "value2",
+}
+```
+
+### Remove list indexes
+
+Use the `remove_list_indices` option to convert the fields from the source map into lists and put the lists into the target field, as shown in the following example:
```

Comment:
"add the lists to" instead of "put the lists into"?

Reply:
See previous comment about put versus add.